testing/deployer/topology/compile.go

```diff
@@ -317,6 +317,13 @@ func compile(logger hclog.Logger, raw *Config, prev *Topology) (*Topology, error
 				return nil, fmt.Errorf("cluster %q node %q has more than one public address", c.Name, n.Name)
 			}
 
+			if n.IsDataplane() && len(n.Services) > 1 {
+				// Our use of consul-dataplane here is supposed to mimic that
+				// of consul-k8s, which ultimately has one IP per Service, so
+				// we introduce the same limitation here.
+				return nil, fmt.Errorf("cluster %q node %q uses dataplane, but has more than one service", c.Name, n.Name)
```

Comment:
Working on the tproxy stuff made it really obvious that a bunch of the stuff here definitely assumes one-dataplane-per-IP, and since the `topology.Node` construct vaguely means `thing with an IP` we can't really have more than one service per node.